src/main/java/org/xstefank/model/PropertiesLoader.java

```diff
@@ -0,0 +1,67 @@
+package org.xstefank.model;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+import static org.xstefank.model.Utils.JBOSS_CONFIG_DIR;
+import static org.xstefank.model.Utils.CONFIG_FILE;
+
+public class PropertiesLoader {
+
+    private static PropertiesLoader propertiesLoader = null;
+    private Properties properties;
+
+    private PropertiesLoader() {
+        String dirProperty = System.getProperty(JBOSS_CONFIG_DIR);
+        if (dirProperty != null) {
+            File dir = new File(System.getProperty(JBOSS_CONFIG_DIR));
+            File targetFile = new File(dir, CONFIG_FILE);
+            this.properties = readPropertiesFromFile(targetFile);
+        }
+    }
+
+    public static synchronized PropertiesLoader getInstance() {
+        if (propertiesLoader == null) {
+            propertiesLoader = new PropertiesLoader();
+        }
+        return propertiesLoader;
+    }
+
+    public Properties getProperties() {
+        if (properties != null)
+            return properties;
+        throw new NullPointerException("Properties were not load properly");
```

Comment:
this is not a way of how to handle errors, you should fail the application when you know the properties haven't been loaded properly, not when somebody is accessing them, this would result into a stactrace that has nothing to do with the actual error